lib/main/transcriptionService.ts

```diff
@@ -47,6 +48,15 @@ export class TranscriptionService {
     this.interactionStartTime = Date.now() // Record start time
     log.info('[gRPC Service] Starting new transcription stream.')
 
+    // Get current interaction ID for trace logging
+    const globalInteractionId = (globalThis as any).currentInteractionId
+    if (globalInteractionId) {
+      traceLogger.logStep(globalInteractionId, 'TRANSCRIPTION_START', {
+        localInteractionId: this.currentInteractionId,
+        startTime: this.interactionStartTime,
+      })
+    }
```

Comment:
Can we make the api of the tracelogger handle whether or not there is a globalinteractionId? It would save on a lot of the boilerplate of getting the global id and then checking if it exists

Reply:
Actually I've changed my mind on this. I don't think we should move this to the traceLogger as then that would tie the traceLogger directly to interactions. This will preclude us from using it for different trace logging in the future.